### PR TITLE
Refactoring to use a ManyToMany field for relationship between Conten…

### PR DESCRIPTION
**Pull request by christopappas**
…tMetadata and CatalogQuery

## Description

Removing CatalogContentKey table in favor of using a ManyToMany field

A snippet from the shell to see what working with this would look like:

```
>>> cq1 = CatalogQuery.objects.first()
>>> cq2 = CatalogQuery.objects.last()
>>> cm1 = ContentMetadata.objects.first()
>>> cm2 = ContentMetadata.objects.last() 
>>> 
>>> cm1.catalog_queries.add(cq1)
>>> cq1.contentmetadata_set.add(cm2)
>>> cq2.contentmetadata_set.add(cm2)
>>> 
>>> cq1.contentmetadata_set.all()
<QuerySet [<ContentMetadata: <ContentMetadata for 'testing'>>, <ContentMetadata: <ContentMetadata for 'testing3'>>]>
>>> cq2.contentmetadata_set.all()
<QuerySet [<ContentMetadata: <ContentMetadata for 'testing3'>>]>
>>> 
>>> cm1.catalog_queries.all()
<QuerySet [<CatalogQuery: <CatalogQuery with content filter hash '8393dd4a16f0e1c9723f9d9cfa2c39f1'>>]>
>>> cm2.catalog_queries.all()
<QuerySet [<CatalogQuery: <CatalogQuery with content filter hash '8393dd4a16f0e1c9723f9d9cfa2c39f1'>>, <CatalogQuery: <CatalogQuery with content filter hash '9a293a221790e7dd818b6c83adb7c98f'>>]>
>>> 
```

## Ticket Link

Link to the associated ticket
[Link title](https://openedx.atlassian.net/browse/ENT-XXXX)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/models.py [hunk 1]
```diff
@@ -159,6 +159,7 @@ class ContentMetadata(TimeStampedModel):
             "endpoint results, specified as a JSON object."
         )
     )
+    catalog_queries = models.ManyToManyField(CatalogQuery)
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py:162]: Do we want a related name here? Something like `content_metadata_chunks` (although I think we can do better than that)

File: enterprise_catalog/apps/catalog/tests/factories.py [hunk 2]
```diff
@@ -9,7 +9,6 @@
     json_serialized_course_modes,
 )
 from enterprise_catalog.apps.catalog.models import (
-    CatalogContentKey,
     CatalogQuery,
     ContentMetadata,
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/tests/factories.py:13]: Nit: This factory could be updated now, but I'm fine leaving it until we actually start using it

**Review by bbaker6225 — APPROVED** [review]

File: enterprise_catalog/apps/catalog/tests/factories.py [hunk 3]
```diff
@@ -59,17 +58,6 @@ class Meta:
     json_metadata = "{}"  # Default metadata to empty object
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/catalog/tests/factories.py:59]: Should we add `catalog_queries` to this factory?

File: enterprise_catalog/apps/catalog/tests/factories.py (hunk 2 above)
Comment (christopappas) [reply enterprise_catalog/apps/catalog/tests/factories.py:13]: So @adamstankiewicz @bbaker6225 , perhaps this is personal preference, but I am of the opinion that factories should not create any related objects when a many-to-many relationship exists, because in theory there could be any number of relationships that could be created by default. Instead, a factory would be used to make any given object, and then the writer of test code would specify the connection between any number of catalog queries and content metadata objects

in other words: I felt like there was no good default to specify with a many-to-many relationship

File: enterprise_catalog/apps/catalog/models.py (hunk 1 above)
Comment (christopappas) [reply enterprise_catalog/apps/catalog/models.py:162]: by default it is `contentmetadata_set`. i'm comfortable with this because it's a pretty common naming pattern, but if others on the team feel there is a more useful related name, i'm happy to update

File: enterprise_catalog/apps/catalog/tests/factories.py (hunk 2 above)
Comment (bbaker6225) [reply enterprise_catalog/apps/catalog/tests/factories.py:13]: Fine with me. If we find that there is a more convenient way to do this factory down the road we can do it then
